nomenclature/testing.py

```diff
@@ -61,6 +62,10 @@ def assert_valid_structure(
         )
     if dimensions is None:
         dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+        if not dimensions:
```

Comment:
You could simply move the new lines outside of the if-statement above, then the empty-check would also apply if a user calls the validation with ` --dimensions "[]"`.

Reply:
OK thank you! That was indeed what I wanted to do but I was unsure we wanted this behaviour or not

Reply:
@danielhuppmann Can I actually separate the case where the sub-folders do not exist and the case when "[]" is given (but sub-folders exist)? I feel like the error message should not be the same...

Reply:
You are absolutely right, these are two separate errors...

Reply:
I took the liberty of doing so, tell me if I should change it